src/core/or/relay.c

```diff
@@ -577,6 +577,14 @@ relay_send_command_from_edge_,(streamid_t stream_id, circuit_t *circ,
   if (payload_len)
     memcpy(cell.payload+RELAY_HEADER_SIZE, payload, payload_len);
 
+  /* Add random bytes to the unused portion of the payload, to foil attacks
+   * where the other side can predict all of the bytes in the payload and thus
+   * compute authenticated sendme cells without seeing the traffic.  See
+   * proposal 289. */
+  crypto_fast_rng_getbytes(get_thread_fast_rng(),
```

Comment:
I recall an old suggestion I made once: let's skip the first 4 bytes of the unused data.  Having some bytes that we know will be set to 0 if they are unused has saved us a lot of times in the past.

Reply:
Done in commit 0c23471a35dfdd4cd286a31ee8b9774008001b71

There is some more calculation to make sure we don't overflow here. Needs special attentions :).